### PR TITLE
Added test for gmm probability

### PR DESCRIPTION
**Pull request by gauravcr7rm**
Added test for gmm probability

### CODE REVIEW

**Comment by gauravcr7rm** [comment]
@rcurtin please see and tell if sample gmm model that i have created is valid or not.
If not that please tell me the way through which i can invoke gmm_train and make a sample gmm model.
Thanks in advance.

**Comment by manish7294** [comment]
@gauravcr7rm Please add the test to cmake file (you can find it at `mlpack/tests/CMakeLists.txt`), so that it can be build. Same goes for #1655.

**Comment by gauravcr7rm** [comment]
I have updated CMakeList to build the tests.
@zoq @rcurtin please tell if some changes are required in the above test.
I have also opened  a pr #1689 for gmm_generate.Please see that as well.

File: src/mlpack/tests/main_tests/gmm_probability.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            << 0 << 4 << 3 << 4 << 8 << 9 << 2 << 5 <<endr;
+
+  SetInputParam("input" , std::move(InputData));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+  
+//Make sure that input points are provided
```
Comment (zoq) [inline src/mlpack/tests/main_tests/gmm_probability.cpp]: Picky style comment, missing space after ``//`` and missing ``.`` at the end, for more details see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#comments.

File: src/mlpack/tests/main_tests/gmm_probability_test.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+{
+  int g = 3;
+  int trials = 2;
+
+
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+  BOOST_FAIL("Unable to load train dataset vc2.csv!");
+  
+  gmm = new GMM((size_t)g , inputdata.n_rows);
```
Comment (zoq) [inline src/mlpack/tests/main_tests/gmm_probability_test.cpp]: Maybe I missed something but I there a reason to use ``new`` here?

Comment (gauravcr7rm) [reply src/mlpack/tests/main_tests/gmm_probability_test.cpp]: Here im trying to initialise a gmm model for which i have used new.

File: src/mlpack/tests/main_tests/gmm_probability.cpp (hunk 1 above)
Comment (gauravcr7rm) [reply src/mlpack/tests/main_tests/gmm_probability.cpp]: I will do the changes suggested

**Comment by rcurtin** [comment]
Oh, sorry, I forgot two things:

1. Can you make sure to add this test to `CMakeLists.txt`?  Right now it is not being tested.
2. Don't worry about the current CI test failure---that doesn't seem to have anything to do with your code.

**Comment by gauravcr7rm** [comment]
@rcurtin what changes are supposed to be done here.

**Comment by rcurtin** [comment]
This is also solved by #1899.

**Pull request closed without merging**